consensus/misc/rewards.go

```diff
@@ -33,6 +33,55 @@ func QuaiToQi(block *types.WorkObject, exchangeRate *big.Int, difficulty *big.In
 	return new(big.Int).Quo(qiByQuai, CalculateQuaiReward(difficulty, exchangeRate))
 }
 
+// ComputeConversionAmountInQuai computes the amount of conversion volume in
+// quai for the given set of newInboundEtxs using the values from the header
+func ComputeConversionAmountInQuai(header *types.WorkObject, newInboundEtxs types.Transactions) *big.Int {
+	conversionAmountInQuai := big.NewInt(0)
+	for _, etx := range newInboundEtxs {
+		// If the etx is conversion
+		if types.IsConversionTx(etx) {
+			value := etx.Value()
+			// If to is in Qi, convert the value into Qi
+			if etx.To().IsInQiLedgerScope() {
+				conversionAmountInQuai = new(big.Int).Add(conversionAmountInQuai, value)
+			}
+			// If To is in Quai, convert the value into Quai
+			if etx.To().IsInQuaiLedgerScope() {
+				value = QiToQuai(header, header.ExchangeRate(), header.MinerDifficulty(), value)
+				conversionAmountInQuai = new(big.Int).Add(conversionAmountInQuai, value)
+			}
+		}
+	}
+	return conversionAmountInQuai
+}
+
+// ApplyCubicDiscount applies the slippage based on the historical
+// converted amounts and apply the
+func ApplyCubicDiscount(valueInt, meanInt *big.Int) *big.Float {
+	value := new(big.Float).SetInt(valueInt)
+	mean := new(big.Float).SetInt(meanInt)
+	tenTimesAverage := new(big.Float).Mul(mean, new(big.Float).SetInt64(10))
+	if value.Cmp(mean) <= 0 {
+		value = new(big.Float).Mul(value, new(big.Float).SetInt64(99))
```

Comment:
we should make 99 and 100 as parameters. I think we should also start this out at 0.2%.

Reply:
okay will multiply by 998 and divide by 1000

Reply:
done